### ISSUE TITLE
Test failures with Cython 3

### ISSUE THREAD

**Issue by musicinmybrain** [mozman/ezdxf#913]
**Describe the bug**
Many tests fail when the extension is compiled with [Cython 3](https://cython.readthedocs.io/en/latest/src/changes.html).

**To Reproduce**

```
python3 -m venv _e
. _e/bin/activate
pip install Cython pytest
python setup.py build_ext
pip install -e .
python -m pytest
```

**Confidential Data**

N/A

**Expected behavior**

The tests should all pass, as they do if `pip install Cython<3' is substituted, or if `python setup.py build_ext` is skipped (pure Python).

**Screenshots**

```
=========================== short test summary info ============================
FAILED tests/test_02_dxf_graphics/test_210_polyline_1.py::test_polyline2d_transform_interface - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_210_polyline_1.py::test_2d_polyline_has_any_start_width - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_210_polyline_1.py::test_2d_polyline_has_any_end_width - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_210_polyline_1.py::test_2d_polyline_has_any_arc - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_210_polyline_explode.py::test_polyline2d_closed - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_210_polyline_explode.py::test_polyline2d_explode - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_210_polyline_vertex_attributes.py::test_vertext_attribs_xy - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_210_polyline_vertex_attributes.py::test_vertext_attribs_xyb - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_210_polyline_vertex_attributes.py::test_vertext_attribs_xyseb - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_210_polyline_vertex_attributes.py::test_append_formatted_vertices - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_215_dimension.py::test_audit_invalid_dimstyle - TypeError: Argument 'z' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_216_dimlines_R12.py::test_horizontal_dimline - TypeError: Argument 'z' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_217_dimlines_R2000.py::test_add_horizontal_dimline - TypeError: Argument 'z' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_217_dimlines_R2000.py::test_virtual_entities_and_explode - TypeError: Argument 'z' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_217_dimlines_R2000.py::test_transformation_of_associated_anonymous_geometry_block - TypeError: Argument 'z' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_217_dimlines_R2000.py::test_copy_dimension_with_geometry_block - TypeError: Argument 'z' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_217_dimlines_R2000.py::test_destroy_virtual_dimension_copy - TypeError: Argument 'z' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_217_dimlines_R2000.py::test_transform_virtual_geometry_block - TypeError: Argument 'z' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_217_dimlines_R2000.py::test_add_virtual_dimension_copy_to_layout - TypeError: Argument 'z' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_217_dimlines_R2000.py::test_supports_virtual_entities_protocol - TypeError: Argument 'z' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_229c_hatch_transformations.py::test_polyline_path_transform_interface - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_229c_hatch_transformations.py::test_edge_path_transform_interface - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_229c_hatch_transformations.py::test_full_circle_ellipse_edge_rotation[arc] - TypeError: Argument 'x' has incorrect type (expected float, got int)   
FAILED tests/test_02_dxf_graphics/test_229c_hatch_transformations.py::test_full_circle_ellipse_edge_rotation[ellipse] - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_229c_hatch_transformations.py::test_full_circle_edge_scaling - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_229c_hatch_transformations.py::test_ocs_mirror_transformations_of_clockwise_oriented_curves[arc-mirror-x] - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_229c_hatch_transformations.py::test_ocs_mirror_transformations_of_clockwise_oriented_curves[arc-mirror-y] - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_229c_hatch_transformations.py::test_ocs_mirror_transformations_of_clockwise_oriented_curves[arc-mirror-xy] - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_229c_hatch_transformations.py::test_ocs_mirror_transformations_of_clockwise_oriented_curves[ellipse-mirror-x] - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_229c_hatch_transformations.py::test_ocs_mirror_transformations_of_clockwise_oriented_curves[ellipse-mirror-y] - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_229c_hatch_transformations.py::test_ocs_mirror_transformations_of_clockwise_oriented_curves[ellipse-mirror-xy] - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_229c_hatch_transformations.py::test_wcs_mirror_transformations_of_clockwise_oriented_curves[arc-mirror-x] - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_229c_hatch_transformations.py::test_wcs_mirror_transformations_of_clockwise_oriented_curves[arc-mirror-y] - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_229c_hatch_transformations.py::test_wcs_mirror_transformations_of_clockwise_oriented_curves[arc-mirror-xy] - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_229c_hatch_transformations.py::test_wcs_mirror_transformations_of_clockwise_oriented_curves[ellipse-mirror-x] - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_229c_hatch_transformations.py::test_wcs_mirror_transformations_of_clockwise_oriented_curves[ellipse-mirror-y] - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_229c_hatch_transformations.py::test_wcs_mirror_transformations_of_clockwise_oriented_curves[ellipse-mirror-xy] - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_229c_hatch_transformations.py::test_wcs_mirror_transformations_for_all_edge_types[mirror-x] - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_229c_hatch_transformations.py::test_wcs_mirror_transformations_for_all_edge_types[mirror-y] - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_229c_hatch_transformations.py::test_wcs_mirror_transformations_for_all_edge_types[mirror-xy] - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_237_mline.py::TestMLine::test_add_two_vertices - TypeError: Argument 'z' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_237_mline.py::TestMLine::test_x_rotation - TypeError: Argument 'z' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_237_mline.py::TestMLine::test_translate - TypeError: Argument 'z' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_237_mline.py::TestMLine::test_uniform_scale - TypeError: Argument 'z' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_237_mline.py::TestMLine::test_non_uniform_scale - TypeError: Argument 'z' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_237_mline.py::TestMLine::test_support_virtual_entities_protocol - TypeError: Argument 'z' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_237_mline.py::TestMLine::test_virtual_sub_entities_source_tracking - TypeError: Argument 'z' has incorrect type (expected float, got int)
FAILED tests/test_02_dxf_graphics/test_239_proxy_graphic.py::TestProxyGraphic::test_can_parse_R2018_lwpolyline - ezdxf.proxygraphic.ProxyGraphicError: Proxy graphic error: Argument 'x' has...
FAILED tests/test_02_dxf_graphics/test_251_upright.py::test_upright_polyline[polyline2d] - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_04_dxf_high_level_structs/test_414_block_reference_content.py::test_skipped_entities_callback - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_04_dxf_high_level_structs/test_416a_disassemble.py::test_2d_polyline_including_width_to_primitive - TypeError: Argument 'x' has incorrect type (expected float, got int) 
FAILED tests/test_05_tools/test_524_block_reference_manager.py::test_count_references_for_anonymous_dimension_block - TypeError: Argument 'z' has incorrect type (expected float, got int) 
FAILED tests/test_05_tools/test_526_explode.py::test_explode_scaled_block_ref_containing_a_hatch - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_06_math/test_602_vec3.py::test_radd_vector[Vec31] - TypeError: can only concatenate tuple (not "ezdxf.acc.vector.Vec3") to tuple
FAILED tests/test_06_math/test_602_vec3.py::test_replace[Vec31] - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_06_math/test_648_construction_ellipse.py::test_to_ocs - TypeError: Argument 'z' has incorrect type (expected float, got int)
FAILED tests/test_07_render/test_704_render_linear_dimension.py::test_dimension_insert_attribute_translates_the_block_content - TypeError: Argument 'z' has incorrect type (expected float, got int)
FAILED tests/test_07_render/test_704_render_linear_dimension.py::test_override_all_colors[1] - TypeError: Argument 'z' has incorrect type (expected float, got int)
FAILED tests/test_07_render/test_704_render_linear_dimension.py::test_override_all_colors[7] - TypeError: Argument 'z' has incorrect type (expected float, got int)
FAILED tests/test_07_render/test_708a_path.py::TestSubPath::test_multi_path_object_to_wcs - TypeError: Argument 'z' has incorrect type (expected float, got int)
FAILED tests/test_07_render/test_708a_path.py::test_polyline_lines - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_07_render/test_708a_path.py::test_polyline_with_bulges - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_07_render/test_708a_path.py::test_from_edge_path - TypeError: Argument 'z' has incorrect type (expected float, got int)
FAILED tests/test_07_render/test_708a_path.py::TestPathFromEdgePathWithElevationAndFlippedExtrusion::test_ellipse_edge - TypeError: Argument 'z' has incorrect type (expected float, got int)
FAILED tests/test_07_render/test_708a_path.py::TestPathFromEdgePathWithElevationAndFlippedExtrusion::test_from_complex_edge_path - TypeError: Argument 'z' has incorrect type (expected float, got int)
FAILED tests/test_07_render/test_712_render_curved_dimension.py::test_dimension_line_divided_by_measurement_text[60-120] - TypeError: Argument 'z' has incorrect type (expected float, got int)
FAILED tests/test_07_render/test_712_render_curved_dimension.py::test_dimension_line_divided_by_measurement_text[300-240] - TypeError: Argument 'z' has incorrect type (expected float, got int)
FAILED tests/test_07_render/test_712_render_curved_dimension.py::test_dimension_line_divided_by_measurement_text[240-300] - TypeError: Argument 'z' has incorrect type (expected float, got int)
FAILED tests/test_07_render/test_712_render_curved_dimension.py::test_dimension_line_divided_by_measurement_text[300-30] - TypeError: Argument 'z' has incorrect type (expected float, got int)
FAILED tests/test_07_render/test_712_render_curved_dimension.py::test_text_and_arrows_fit_between_extension_lines - TypeError: Argument 'z' has incorrect type (expected float, got int)   
FAILED tests/test_07_render/test_712_render_curved_dimension.py::test_has_outside_text_and_arrows[3] - TypeError: Argument 'z' has incorrect type (expected float, got int)
FAILED tests/test_07_render/test_712_render_curved_dimension.py::test_has_outside_text_and_arrows[6] - TypeError: Argument 'z' has incorrect type (expected float, got int)
FAILED tests/test_07_render/test_712_render_curved_dimension.py::test_has_outside_text_and_arrows_but_not_a_wide_text - TypeError: Argument 'z' has incorrect type (expected float, got int)
FAILED tests/test_07_render/test_712_render_curved_dimension.py::test_fixed_length_extension_lines - TypeError: Argument 'z' has incorrect type (expected float, got int)
FAILED tests/test_08_addons/test_811a_drawing_frontend.py::test_banded_polyline_2d - TypeError: Argument 'x' has incorrect type (expected float, got int)
FAILED tests/test_10_issues/test_issue_414_bbox_calculation.py::TestMakePath::test_make_path - TypeError: Argument 'z' has incorrect type (expected float, got int)
FAILED tests/test_10_issues/test_issue_414_bbox_calculation.py::TestMakePath::test_bounding_box_calculation - TypeError: Argument 'z' has incorrect type (expected float, got int)
ERROR tests/test_02_dxf_graphics/test_210_polyline_explode.py::test_polyline2d_virtual_entities - TypeError: Argument 'x' has incorrect type (expected float, got int)
ERROR tests/test_02_dxf_graphics/test_210_polyline_explode.py::test_virtual_sub_entities_source_tracking - TypeError: Argument 'x' has incorrect type (expected float, got int)
ERROR tests/test_02_dxf_graphics/test_210_polyline_explode.py::test_polyline2d_elevation - TypeError: Argument 'x' has incorrect type (expected float, got int)
ERROR tests/test_02_dxf_graphics/test_237_mline.py::TestMLineAudit::test_valid_mline - TypeError: Argument 'z' has incorrect type (expected float, got int)
ERROR tests/test_02_dxf_graphics/test_237_mline.py::TestMLineAudit::test_fix_invalid_style_name - TypeError: Argument 'z' has incorrect type (expected float, got int)
ERROR tests/test_02_dxf_graphics/test_237_mline.py::TestMLineAudit::test_fix_invalid_style_handle - TypeError: Argument 'z' has incorrect type (expected float, got int)
ERROR tests/test_02_dxf_graphics/test_237_mline.py::TestMLineAudit::test_fix_invalid_style_handle_by_name - TypeError: Argument 'z' has incorrect type (expected float, got int)
ERROR tests/test_02_dxf_graphics/test_237_mline.py::TestMLineAudit::test_fix_invalid_line_direction - TypeError: Argument 'z' has incorrect type (expected float, got int)
ERROR tests/test_02_dxf_graphics/test_237_mline.py::TestMLineAudit::test_fix_invalid_miter_direction - TypeError: Argument 'z' has incorrect type (expected float, got int)
ERROR tests/test_02_dxf_graphics/test_237_mline.py::TestMLineAudit::test_fix_invalid_line_parameters - TypeError: Argument 'z' has incorrect type (expected float, got int)
ERROR tests/test_02_dxf_graphics/test_250_virtual_boundary_path_entities.py::TestSpatialTransformation::test_wcs_line - TypeError: Argument 'x' has incorrect type (expected float, got int)
ERROR tests/test_02_dxf_graphics/test_250_virtual_boundary_path_entities.py::TestSpatialTransformation::test_wcs_spline - TypeError: Argument 'x' has incorrect type (expected float, got int)
ERROR tests/test_02_dxf_graphics/test_250_virtual_boundary_path_entities.py::TestSpatialTransformation::test_ocs_clockwise_arc - TypeError: Argument 'x' has incorrect type (expected float, got int)
ERROR tests/test_02_dxf_graphics/test_250_virtual_boundary_path_entities.py::TestSpatialTransformation::test_ocs_counter_clockwise_arc - TypeError: Argument 'x' has incorrect type (expected float, got int)
ERROR tests/test_02_dxf_graphics/test_250_virtual_boundary_path_entities.py::TestSpatialTransformation::test_wcs_clockwise_ellipse - TypeError: Argument 'x' has incorrect type (expected float, got int)
ERROR tests/test_02_dxf_graphics/test_250_virtual_boundary_path_entities.py::TestSpatialTransformation::test_wcs_counter_clockwise_ellipse - TypeError: Argument 'x' has incorrect type (expected float, got int)
ERROR tests/test_05_tools/test_535_xref_basic.py::TestDimension::test_load_dimension_style_exist - TypeError: Argument 'z' has incorrect type (expected float, got int)
ERROR tests/test_05_tools/test_535_xref_basic.py::TestDimension::test_loaded_geometry_block_exist - TypeError: Argument 'z' has incorrect type (expected float, got int)
ERROR tests/test_05_tools/test_535_xref_basic.py::TestDimensionDimStyleOverride::test_load_dimension_style_exist - TypeError: Argument 'z' has incorrect type (expected float, got int)
ERROR tests/test_05_tools/test_535_xref_basic.py::TestDimensionDimStyleOverride::test_dot_blocks_in_source_doc - TypeError: Argument 'z' has incorrect type (expected float, got int)
ERROR tests/test_05_tools/test_535_xref_basic.py::TestDimensionDimStyleOverride::test_loaded_geometry_block_has_two_block_refs_of_dot - TypeError: Argument 'z' has incorrect type (expected float, got int)
ERROR tests/test_05_tools/test_535_xref_basic.py::TestDimensionDimStyleOverride::test_loaded_xdata_override_has_handle_to_existing_block - TypeError: Argument 'z' has incorrect type (expected float, got int)
ERROR tests/test_05_tools/test_535_xref_basic.py::TestLoadMLine::test_loaded_infrastructure - TypeError: Argument 'z' has incorrect type (expected float, got int)
ERROR tests/test_05_tools/test_535_xref_basic.py::TestLoadMLine::test_loaded_mline_has_correct_style_attributes - TypeError: Argument 'z' has incorrect type (expected float, got int)
ERROR tests/test_05_tools/test_535_xref_basic.py::TestLoadMLine::test_loaded_mline_has_same_vertices - TypeError: Argument 'z' has incorrect type (expected float, got int)
ERROR tests/test_07_render/test_715_issue_747_explode_3d_dim.py::test_exploded_arrow_blocks - TypeError: Argument 'z' has incorrect type (expected float, got int)
ERROR tests/test_07_render/test_715_issue_747_explode_3d_dim.py::test_exploded_arc - TypeError: Argument 'z' has incorrect type (expected float, got int)
ERROR tests/test_07_render/test_715_issue_747_explode_3d_dim.py::test_exploded_mtext - TypeError: Argument 'z' has incorrect type (expected float, got int)
ERROR tests/test_07_render/test_715_issue_747_explode_3d_dim.py::test_exploded_extension_line_1 - TypeError: Argument 'z' has incorrect type (expected float, got int)
ERROR tests/test_07_render/test_715_issue_747_explode_3d_dim.py::test_exploded_extension_line_2 - TypeError: Argument 'z' has incorrect type (expected float, got int)
=========== 77 failed, 7069 passed, 33 skipped, 30 errors in 14.48s ============
```

**Comment by mozman** [comment]
fixed in f805a73, but typo in commit message

**Comment by musicinmybrain** [comment]
Thank you! I cherry-picked https://github.com/mozman/ezdxf/commit/f805a73b188a5afc1f01ccf1605ba36ea012a77e to `v1.0.3` and applied it to the [Fedora Linux `python-ezdxf`](https://src.fedoraproject.org/rpms/python-ezdxf) package, and all the tests pass.
